### PR TITLE
feat: support dynamic region for azurestack 

### PR DESCRIPTION
**Pull request by honcao**
**Reason for Change**:

feat:support dynamic region for azurestack 

**Issue Fixed**:

#282

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by mboersma** [comment]
@honcao could you `git commit --amend` the first commit so there's a space after the colon? The tools don't recognize the format otherwise.

```
feat: support dynamic region for azurestack
```

File: pkg/api/apiloader.go [hunk 1]
```diff
@@ -205,6 +205,9 @@ func (a *Apiloader) LoadContainerService(
 		if containerService.Properties == nil {
 			return nil, errors.New("missing ContainerService Properties")
 		}
+		if containerService.Properties.IsAzureStackCloud() && containerService.Location == "" {
```
Comment (jackfrancis) [inline pkg/api/apiloader.go]: Let's move this validation into `pkg/api/vlabs/validate.go`

Comment (jackfrancis) [reply pkg/api/apiloader.go]: The other validations in this surface are tech debt, we will (not in this PR) move those into `validate.go` as well

Comment (honcao) [reply pkg/api/apiloader.go]: I also put the following code into validate.go
		if containerService.Properties == nil {
			return nil, errors.New("missing ContainerService Properties")
		}

**Comment by jackfrancis** [comment]
lgtm pending E2E

thanks @honcao!

**Pull request merged**